SalesforceCommerceCloud/pwa-kit
# @W-14244624@ Add a phased launch hybrid option to generator (WIP)

To enable automated e2e testing of a hybrid PWA / SFRA site, we are adding a new preset to the app generator that generates a PWA that is ready to be used in a hybrid deploy.

This preset applies the PWA changes to routes.jsx documented here: https://developer.salesforce.com/docs/commerce/pwa-kit-managed-runtime/guide/phased-headless-rollouts.html

Assumptions:
* The hybrid app is extensible
* We are redirecting carts and checkout to SFRA. PLP, PDP, login and other pages will be served by PWA

# Types of Changes

* Add a new `retail-react-app-hybrid-test-project` preset to pwa-kit-create-app
* Modify the route.jsx template to apply steps outlined in https://developer.salesforce.com/docs/commerce/pwa-kit-managed-runtime/guide/phased-headless-rollouts.html
  * Modifies the useEffect code that is suggested by the docs to use location and history rather than window


> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# How to Test-Drive This PR

- Checkout this code
- Run `node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --preset retail-react-app-hybrid-test-project`
- Verify that an app is generated
- Build and run the generated app. Verify that it works. Verify that trying to navigate to /cart or /checkout triggers redirects (will return a page not found unless you have hybrid SFRA / ecdn / composable-dev-server set up)

- Also run the generator without this preset
- Verify that the other extensible retail react app presets are not affected by this change (check router.jsx)

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
